main.cpp

```diff
@@ -31,25 +31,22 @@ struct T
 
 struct Comparator                                //4
 {
-    T* compare(T* a = nullptr, T* b = nullptr)//5
+    T compare(T& a, T& b)//5
     {
-        if (a != nullptr && b != nullptr)
-        {
-            if(a->value < b->value) return a;
-            if(a->value > b->value) return b;
-        }
-        return nullptr;   
+        if(a.value < b.value) return a;
+        if(a.value > b.value) return b;
+        T noneOfThem(0, "none of them");
+        return noneOfThem;
```

Comment:
Nope, re-read the instructions about functions that return nullptr.
- [ ] revise